packages/backend-common/src/service/lib/ServiceBuilderImpl.ts

```diff
@@ -120,34 +154,96 @@ export class ServiceBuilderImpl implements ServiceBuilder {
         reject(e);
       });
 
-      const server = stoppable(
-        app.listen(port, host, () => {
-          logger.info(`Listening on ${host}:${port}`);
-        }),
-        0,
-      );
+      let server: http.Server;
+
+      if (httpsSettings) {
+        logger.info('Initializing https server');
+
+        let credentials: { key: string; cert: string };
+        const signingOptions: any = httpsSettings?.certificate;
+
+        if (signingOptions?.algorithm !== undefined) {
+          const certificateAttributes: Array<any> = [];
+
+          Object.getOwnPropertyNames(signingOptions?.attributes).forEach(
```

Comment:
You can use `Object.entries(signingOptions.attributes).map(([name, value]) => ({ name, value }))` here instead.